test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/PackageSearchRunnerTests.cs

```diff
@@ -0,0 +1,389 @@
+                    }}
+                }}";
+
+            mockServer.Get.Add("/v3/index.json", r => index);
+            mockServer.Get.Add("/search/query?q=json&skip=0&take=20&prerelease=false&semVerLevel=2.0.0", r => _onePackageQueryResult);
+            mockServer.Start();
+
+            // Act
+            await PackageSearchRunner.RunAsync(
+                sourceProvider: sourceProvider,
+                packageSearchArgs,
+                cancellationToken: System.Threading.CancellationToken.None);
+
+            //stop mock server
+            mockServer.Stop();
+
+            // Assert
+            foreach (var expected in expectedValues)
+            {
+                Assert.Contains(expected, _fixture.ColoredMessage.Select(tuple => tuple.Item1));
+            }
+
+            Assert.Contains(Tuple.Create("Json", ConsoleColor.Red), _fixture.ColoredMessage);
+        }
+
+        [Theory]
+        [InlineData(0, 10, true)]
+        [InlineData(0, 20, false)]
+        [InlineData(5, 10, true)]
+        [InlineData(10, 20, false)]
+        public async Task PackageSearchRunner_SearchAPIWithVariousSkipTakePrereleaseOptionsValuesReturnsOnePackage_OnePackageTableOutputted(int skip, int take, bool prerelease)
+        {
+            // Arrange
+            ISettings settings = Settings.LoadDefaultSettings(Directory.GetCurrentDirectory(),
+                    configFileName: null,
+                    machineWideSettings: new XPlatMachineWideSetting());
+            PackageSourceProvider sourceProvider = new PackageSourceProvider(settings);
+            var mockServer = new MockServer();
+            var expectedValues = new List<string>
+                {
+                    "| Package ID           ",
+                    "| Latest Version ",
+                    "| Authors           ",
+                    "| Downloads   ",
+                    "|----------------------",
+                    "|----------------",
+                    "|-------------------",
+                    "|-------------",
+                    "| ",
+                    "",
+                    "Fake.Newtonsoft.",
+                    "Json",
+                    "",
+                    " ",
+                    "| 12.0.3         ",
+                    "| James Newton-King ",
+                    "| 531,607,259 ",
+                };
+
+            PackageSearchArgs packageSearchArgs = new()
+            {
+                Skip = skip,
+                Take = take,
+                Prerelease = prerelease,
+                ExactMatch = false,
+                Logger = _fixture.GetLogger(),
+                SearchTerm = "json",
+                Sources = new List<string> { $"{mockServer.Uri}v3/index.json" }
+            };
+
+            string index = $@"
+                {{
+                    ""version"": ""3.0.0"",
+
+                    ""resources"": [
+                    {{
+                        ""@id"": ""{mockServer.Uri + "search/query"}"",
```

Comment:
I was hoping a utility method I made for `push` tests would be useful to you, but after reviewing, it's only creating the index endpoint. I think it'd take some refactoring (and moving to a util method) to use that here. 
If you agree, then it's something we can work on (together if you like) separately from this PR. 
https://github.com/NuGet/NuGet.Client/blob/452ef85379dd7605387f8bfab0c7ad1c856690a1/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs#L1007

Reply:
I have updated the tests to contain one mockserver and one index port.